src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -251,24 +250,52 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   (moduleConfig as any).preloadPlugins = [];
 
   let resourcesLoaded = 0;
-  function setProgress(){
-      resourcesLoaded++;
-      const percentage = resourcesLoaded / totalResources.length * 100;
-      document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
-      document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  function setProgress() {
+    resourcesLoaded++;
+    const percentage = resourcesLoaded / totalResources.length * 100;
+    document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
+    document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  }
+
+  const typesMap: { [key: string]: WebAssemblyBootResourceType | undefined } = {
+    'assembly': 'assembly',
+    'pdb': 'pdb',
+    'icu': 'globalization',
+    'dotnetwasm': 'dotnetwasm',
+  };
+
+  // it would not `loadResource` on types for which there is no typesMap mapping
+  const downloadResource = (request: ResourceRequest): LoadingResource | undefined => {
+    const type = typesMap[request.behavior];
+    if (type !== undefined) {
+      const loaded = resourceLoader.loadResource(request.name, request.resolvedUrl!, request.hash!, type);
+      loaded.url = toAbsoluteUrl(loaded.url);
+      return loaded;
     }
+    return undefined;
+  };
+
+  const runtimeAssets = resourceLoader.bootConfig.resources.runtimeAssets;
+  // pass part of responsibility for asset loading to runtime
+  const assets: AssetEntry[] = Object.keys(runtimeAssets).map(name => {
+    const asset = runtimeAssets[name] as AssetEntry;
+    asset.name = name;
+    asset.resolvedUrl = `_framework/${name}`;
+    return asset;
+  });
+
+  // blazor could start downloading bit earlier than the runtime would
+  const assetsBeingLoaded = assets
```

Comment:
This name doesn't seem right. We're loading lots of assets, and this is only a subset of them. If we're filtering to only the `dotnetwasm` entry, can the name reflect that? For example `dotnetWasmAssetsBeingLoaded`.

Just trying to make it more practical for people to read this in the future :)

Reply:
The filter would look differently in net8, so `runtimeAssetsBeingLoaded`